gluten-core/src/main/scala/io/glutenproject/extension/ColumnarOverrides.scala

```diff
@@ -560,26 +566,96 @@ case class ColumnarOverrideRules(session: SparkSession)
       overridden
     }
 
+  def fallbackWholeStage(plan: SparkPlan): Boolean = {
+    if (wholeStageFallbackThreshold == -1) {
+      return false
+    }
+    var fallbacks = 0
+    def countFallback(plan: SparkPlan): Unit = {
+      plan match {
+        // Another stage.
+        case _: QueryStageExec =>
+          return
+        case ColumnarToRowExec(_: GlutenPlan) =>
+          fallbacks = fallbacks + 1
+        // Possible fallback for leaf node.
+        case leafPlan: LeafExecNode if !leafPlan.isInstanceOf[GlutenPlan] =>
+          fallbacks = fallbacks + 1
+        case _ =>
+      }
+      plan.children.map(p => countFallback(p))
+    }
+    countFallback(plan)
+    if (fallbacks >= wholeStageFallbackThreshold) {
+      true
+    } else {
+      false
+    }
```

Comment:
Could just be replaced with `fallbacks >= wholeStageFallbackThreshold` and remove the if-else clause.

Reply:
Fixed. Thanks!